src/docfx/Errors.cs

```diff
@@ -4,174 +4,390 @@
 
+        /// <summary>
+        /// Docfx.yml/docfx.json doesn't exist at the repo root.
+        /// Examples:
+        ///   - non-loc build, docfx.yml/docfx.json doesn't exist at the repo root
+        ///   - loc build, docfx.yml/docfx.json doesn't exist at the neither loc repo root
+        ///     nor source repo root
+        /// </summary>
         public static Error ConfigNotFound(string docsetPath)
             => new Error(ErrorLevel.Error, "config-not-found", $"Cannot find 'docfx.yml/docfx.json' at '{docsetPath}'");
 
+        /// <summary>
+        /// Two files include each other.
+        /// Examples:
+        ///   - a.md includes b.md and reversely
+        ///   - toc1 references toc2 and reversely
+        ///   - toc references itself
+        /// </summary>
         public static Error CircularReference<T>(IEnumerable<T> dependencyChain)
             => new Error(ErrorLevel.Error, "circular-reference", $"Found circular reference: {string.Join(" --> ", dependencyChain.Select(file => $"'{file}'"))}");
 
+        /// <summary>
+        /// Didn't run `docfx restore` before running `docfx build`.
+        /// Examples:
+        /// - can't find a cache(build required) file defined with url in config file
+        /// - can't find dependent repo in file system
+        /// </summary>
         public static Error NeedRestore(string dependenyRepoHref)
             => new Error(ErrorLevel.Error, "need-restore", $"Cannot find dependency '{dependenyRepoHref}', did you forget to run `docfx restore`?");
 
+        /// <summary>
+        /// Failed to get a user from neither user cache nor github api by login.
+        /// Examples:
+        ///   - defined a non-existent author
+        /// </summary>
         public static Error GitHubUserNotFound(string login)
             => new Error(ErrorLevel.Warning, "github-user-not-found", $"Cannot find user '{login}' on GitHub");
 
+        /// <summary>
+        /// Failed to call a github api, e.g. GET /users/login.
+        /// Examples:
+        ///   - the api call reach github limit
+        ///   - using invalid access token(more detailed info in ex.Message)
+        /// </summary>
         public static Error GitHubApiFailed(string api, Exception ex)
             => new Error(ErrorLevel.Warning, "github-api-failed", $"Failed calling GitHub API '{api}': {ex.Message}");
 
+        /// <summary>
+        /// In yaml-format toc, topicHref should reference an article,
+        /// rather than relative path or another toc file.
+        /// </summary>
         public static Error InvalidTopicHref(Document relativeTo, string topicHref)
             => new Error(ErrorLevel.Error, "invalid-topic-href", $"The topic href '{topicHref}' can only reference to a local file or absolute path", relativeTo.ToString());
 
+        /// <summary>
+        /// In markdown-format toc, link(treated as inclusion) can only be toc file, folder or absolute path.
```

Comment:
minor: you can use upper case for SHOULD/CAN/ONLY, as most descriptions here are about "what is wrong", only `InvalidTopicHref` and `InvalidTocHref` are saying "what is right"